projects/swimlane/ngx-ui/src/lib/components/json-editor/json-editor-flat/json-editor-node-flat/node-types/object-node-flat/object-node-flat.component.ts

```diff
@@ -82,16 +82,27 @@ export class ObjectNodeFlatComponent extends ObjectNode implements OnInit, OnCha
   }
 
   onUpdatePropertyName(options: { id: string; name: string }): void {
-    const existingSchemaProperty = this.schemaRef.properties[options.name];
+    let schema = this.schemaBuilderMode ? this.schemaRef : this.schema;
+
+    if (!schema?.properties) {
+      if (!schema) {
+        schema = {};
+      }
+      schema.properties = {};
+    }
+
+    const existingSchemaProperty = schema.properties[options.name];
     const existingPropertyValue = this.model[options.name];
     const oldName = this.propertyIndex[options.id].propertyName;
 
     this.duplicatedFields.delete(options.id);
 
     if (!existingSchemaProperty && existingPropertyValue === undefined) {
-      const index = Object.keys(this.schemaRef.properties).findIndex(prop => prop === oldName);
-      this.updateSchemaPropertyName(this.schemaRef, options.name, this.propertyIndex[options.id].propertyName);
-      this.swapSchemaProperties(index);
+      const index = Object.keys(schema.properties).findIndex(prop => prop === oldName);
```

Comment:
Is this the same as `Object.keys(schema.properties).indexOf(oldName)`?

Reply:
it is yep, changed it.